AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java

```diff
@@ -272,7 +272,24 @@ public void load() {
 
     public String loadColumn(String columnName) {
         int pos = 1;
-        int chunk = 1024*1024; // 1 mb, a little less than what a cursor line may contain
+        // the window size is saved in
+        // io.requery.android.database.CursorWindow.sCursorWindowSize.
+        // Values are copied here. Ideally, a getter would allow to access it.
+        final int WINDOW_SIZE_KB = 2048;
+        final int sCursorWindowSize = WINDOW_SIZE_KB * 1024;
+        // reduce the actual size a little bit.
+        // We have the ability to look into our sqlite implementation on Android and use it's value
+        // as a ceiling. Try it, with a reasonable fallback in case of failure
+        if ("io.requery.android.database.sqlite.SQLiteDatabase".equals(mDb.getDatabase().getClass().getName())) {
```

Comment:
This isn't going to work now #3524 has gone in as it applies a decorator to the class

Reply:
We need to peek through the decorator I think? We need to be able to get the real implementation class so we have reasonable assurance that an otherwise known reflective access will work

Reply:
@Arthur-Milchior - @david-allison-1 is right on this one, we'll need a change to the new wrapper as well, in order to access this https://github.com/david-allison-1/Anki-Android/blob/677831e94aa2cbd83284a305cf9cb88050e78f4f/AnkiDroid/src/main/java/com/ichi2/utils/DatabaseChangeDecorator.java#L39 - I believe the change is to make a getter for the wrapped object, then in this PR do a cast in order to access it

```suggestion
        if ("io.requery.android.database.sqlite.SQLiteDatabase".equals(((DatabaseChangeDecorator)mDb.getDatabase()).getWrapped().getClass().getName())) {
```

This PR has been close but refuses to die! I think this last bit will get it